### ISSUE TITLE
Qt Translations are not being copied

### ISSUE THREAD

**Issue by Ribtoks** [probonopd/linuxdeployqt#115]
Environment: [Ubuntu16.04], [Qt 5.6.2] official distribution (from _download.qt.io_).

`windeployqt` and `macdeployqt` somehow manage to pack qt's translations (like _qt_en.qm_) into the deployment location `translations/`, but `linuxdeployqt` does not do so, being inconsistent with the former ones.

**Comment by probonopd** [comment]
Thanks @Ribtoks for reporting this. Since `linuxdeployqt` was originally based on the `macdeployqt` code, do you think you could port the missing/defective part over and send a Pull Request?

**Comment by Ribtoks** [comment]
@probonopd I might look into the translations issue later but I can't promise anything. For the time being it's easier for me just to `grep` & `copy` .qm files directly.

**Comment by probonopd** [comment]
Seems like we have inherited https://bugreports.qt.io/browse/QTBUG-18662

> macdeployqt doesn't copy translation files

**Comment by Ribtoks** [comment]
@probonopd yep, I checked source code of _macdeployqt_ today and found out that it doesn't copy the translations, but _windeployqt_ does. I've added support for that in my fork. Will create a PR later if my changes will work fine.

**Comment by probonopd** [comment]
Regarding the translations, Subsurface seems to search for

```
me@host:~$ strace -f ./squashfs-root/AppRun 2>&1 | grep trans | cut -d '"' -f 2
/home/me/squashfs-root/usr/translations/qt_en.qm
/home/me/squashfs-root/usr/translations/qt_en.qm
/home/me/squashfs-root/usr/translations/qt_en.qm
/home/me/squashfs-root/usr/translations/subsurface_en.qm
/home/me/squashfs-root/usr/translations/subsurface_en
/home/me/squashfs-root/usr/translations/subsurface_en_US.qm
/home/me/squashfs-root/usr/translations/subsurface_en_US
/home/me/squashfs-root/usr/translations/subsurface_en_us.qm
/home/me/squashfs-root/usr/translations/subsurface_en_us
/home/me/squashfs-root/usr/translations/subsurface_en_Latn_US.qm
/home/me/squashfs-root/usr/translations/subsurface_en_Latn_US
/home/me/squashfs-root/usr/translations/subsurface_en_latn_us.qm
/home/me/squashfs-root/usr/translations/subsurface_en_latn_us
/home/me/squashfs-root/usr/translations/subsurface_en.qm
/home/me/squashfs-root/usr/translations/subsurface_en
/home/me/squashfs-root/usr/translations/subsurface_en.qm
/home/me/squashfs-root/usr/translations/subsurface_en
/home/me/squashfs-root/usr/translations/subsurface_en_Latn.qm
/home/me/squashfs-root/usr/translations/subsurface_en_Latn
/home/me/squashfs-root/usr/translations/subsurface_en.qm
/home/me/squashfs-root/usr/translations/subsurface_en
/home/me/squashfs-root/usr/translations/subsurface_en_latn.qm
/home/me/squashfs-root/usr/translations/subsurface_en_latn
/home/me/squashfs-root/usr/translations/subsurface_en.qm
/home/me/squashfs-root/usr/translations/subsurface_en
/home/me/squashfs-root/usr/translations/subsurface_
/home/me/squashfs-root/usr/translations/subsurface
/home/me/squashfs-root/usr/bin/translations
/home/me/squashfs-root/usr/share/subsurface/translations
/home/me/squashfs-root/usr/bin/../Resources/share/translations
/home/me/squashfs-root/usr/bin/../Resources/translations
/home/me/squashfs-root/usr/bin/../translations
/home/me/squashfs-root/usr/translations/subsurface_en.qm
```

Currently we are shipping `qt_en.qml` (albeit in the __wrong location__, possibly a `linuxdeployqt` bug) but __not__ `subsurface_en.qm`. 

Reference:
https://travis-ci.org/probonopd/subsurface/builds/297349863#L5031-L5050

**Comment by probonopd** [comment]
Ping @ribtoks can you please have a look at the above? Thanks.

**Comment by ribtoks** [comment]
@probonopd Hey. I don't know the root cause, but from your code you will end up in `<AppDir>/translations` in `fhsLikeMode`. 

What you have is `appDirInfo.binaryPath` + `/../` + `/translations` for `fhsLikeMode` and for the other you have directly `appDirInfo.path + "/translations"`. Former (if binary path ends up in `<AppDir>/bin`) will be equal to the latter.

Does that ring any bell? Or I'm just stating obvious things?

**Comment by probonopd** [comment]
> @probonopd Hey. I don't know the root cause, but from your code you will end up in <AppDir>/translations in fhsLikeMode.

Which is wrong, it needs to end up in `<AppDir>/<whatever>/translations` in fhsLikeMode, with "whatever" most likely (but not always) being `usr/`.

https://github.com/probonopd/linuxdeployqt/blob/74580fa63e0ee66e2f4c437a1805f2bed69d9a83/tools/linuxdeployqt/shared.cpp#L1195-L1205

is correct for both modes whereas

https://github.com/probonopd/linuxdeployqt/blob/74580fa63e0ee66e2f4c437a1805f2bed69d9a83/tools/linuxdeployqt/shared.cpp#L1612

is wrong for FHS-like mode.

Does this clarify it enough so that you can send a PR?

**Comment by probonopd** [comment]
> UPD. I suppose you assume appDirInfo.binaryPath leads to .../usr/bin/

Correct. But `usr/` could also be called something else, hence it's "the directory above the `bin/` directory.

> UPD 2 Why is Subsurface looking for translations in /usr/translations (strange path)? Is it some environment variable?

The directory above the ´bin/` directory is where `linuxdeployqt` deploys all the Qt subdirectories to in FHS-like mode.

**Comment by ribtoks** [comment]
@probonopd 
Well, yes - you clarified it enough for me to understand (thanks!), but regarding a PR few disclaimers:

1. I have no freaking idea what FHS is. It was the first time I saw this acronym in `linuxdeployqt` codebase
2. I don't want to sound rude, but I guess I didn't write the code which looks for a translation path depending on FHS (see (1)) so I don't have to fix that. I copied the code from `windeployqt` which combines Qt translations for deployed modules and deploys combined one along with the app.

**Comment by probonopd** [comment]
[Filesystem Hierarchy Standard](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard).

Re 2., I know, but nevertheless would greatly appreciate if you could have a go at it as I am rather overloaded at the moment and iirc you had thankfully added the translations functionality.

Thanks for your contributions!

**Comment by ribtoks** [comment]
@probonopd https://github.com/probonopd/linuxdeployqt/pull/176 - this is all I could do right now (if I understood you correctly).

I'm spending all my free time for [Xpiks](https://github.com/ribtoks/xpiks) so please don't ask me for more. Thanks for understanding.

**Comment by probonopd** [comment]
Thank you very much, looks promising. Will test it soon.
